theano/gof/op.py

```diff
@@ -694,6 +694,9 @@ def __ne__(self, other):
     # Python implementation #
     #########################
 
+    def grad2(self, inputs, outputs, output_grads):
```

Comment:
As for a name for grad2, I don't have a rigorous mathematical name. I can suggest "grad_chain", since it's essentially the gradient of the op times the previous outputs from the chain rule.